### PR TITLE
Remove scope id as a required param for worker creation

### PR DESCRIPTION
**Pull request by irenarindos**

### CODE REVIEW

File: docs/resources/worker.md [hunk 1]
```diff
@@ -15,12 +15,9 @@ The resource allows you to create a self-managed worker object.
 <!-- schema generated by tfplugindocs -->
 ## Schema
 
-### Required
-
-- `scope_id` (String) The scope for the worker.
-
 ### Optional
 
+- `scope_id` (String) The scope for the worker.
```
Comment (johanbrandhorst) [inline docs/resources/worker.md:20]: Should we document what happens if this is not set? Does it get created in the global scope?

**Review by louisruch — COMMENTED** [review]
It seems like the unit tests are not happy with this update

File: docs/resources/worker.md (hunk 1 above)
Comment (louisruch) [reply docs/resources/worker.md:20]: Curious if a `scope_id` is optional, my understanding was that it was always global? If that is the case shouldn't we just deprecate this field

Comment (irenarindos) [reply docs/resources/worker.md:20]: Right now it defaults to global, though there has been discussion about allowing workers to exist at other scopes so I'm not sure if we should deprecate it?

Comment (louisruch) [reply docs/resources/worker.md:20]: Na I think leave it in place if it is something we might be supporting in the future. Another option can be to set a default value of global via the provider

File: internal/provider/worker.go [hunk 2]
```diff
@@ -214,7 +214,7 @@ func resourceWorkerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 			return diag.Errorf("error updating worker: %v", err)
 		}
 	}
-
+
```
[the remark this answers is not in the document]
Comment (irenarindos) [reply internal/provider/worker.go]: Done, thanks!

**Pull request merged**